tests/unit_tests/dft/include/compute_out_of_place.hpp

```diff
@@ -0,0 +1,146 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_COMPUTE_OUT_OF_PLACE_HPP
+#define ONEMKL_COMPUTE_OUT_OF_PLACE_HPP
+
+#include "compute_tester.hpp"
+
+/* Note: There is no implementation for Domain Real */
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+int DFT_Test<precision, domain>::test_out_of_place_buffer() {
+    if (!init(TestType::buffer)) {
+        return test_skipped;
+    }
+
+    descriptor_t descriptor{ size };
+    descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                         oneapi::mkl::dft::config_value::NOT_INPLACE);
+    commit_descriptor(descriptor, sycl_queue);
+
+    std::vector<OutputType> out_host(size);
+    std::vector<InputType> out_host_back(size);
+    sycl::buffer<InputType, 1> in_dev{ sycl::range<1>(size) };
+    sycl::buffer<OutputType, 1> out_dev{ sycl::range<1>(size) };
+    sycl::buffer<InputType, 1> out_back_dev{ sycl::range<1>(size) };
```

Comment:
Suggested renaming: `fwd_data_dev`, `bwd_data_dev`, `roundtrip_data_dev` for these three. (Same for USM tests here below, if you agree to the suggestion)

Reply:
Done in [b18bbe6](https://github.com/oneapi-src/oneMKL/pull/261/commits/b18bbe6b2e650d779e2f389ff86f80afea6799d7).